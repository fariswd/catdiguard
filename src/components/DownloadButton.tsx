import React from 'react';
import {TouchableOpacity, PermissionsAndroid, ToastAndroid} from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons';
import CameraRoll from '@react-native-community/cameraroll';

type Props = {
  imageRef: any;
};

export default function DownloadButton(props: Props) {
  const checkPermission = async () => {
    const permission = PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE;
    const hasPermission = await PermissionsAndroid.check(permission);
    if (hasPermission) {
      return true;
    }
    const status = await PermissionsAndroid.request(permission);
    return status === 'granted';
  };
  const saveImage = async () => {
    const isAllowed = await checkPermission();
    if (isAllowed) {
      console.log(props.imageRef.current);
      props.imageRef.current.capture().then(async (uri: string) => {
        await CameraRoll.save(uri, {type: 'photo', album: 'Cat'});
        ToastAndroid.showWithGravity(
          'Cat has been saved!',
          ToastAndroid.SHORT,
          ToastAndroid.BOTTOM,
        );
      });
    }
  };

  return (
    <TouchableOpacity onPress={() => saveImage()}>
      <Ionicons name="download-outline" size={28} color="#666" />
    </TouchableOpacity>
  );
}
