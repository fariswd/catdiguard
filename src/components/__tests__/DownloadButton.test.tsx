import React from 'react';
import DownloadButton from '../DownloadButton';

import renderer from 'react-test-renderer';

describe('Download Button Component', () => {
  it('Should render without crashing', () => {
    const rendered = renderer
      .create(<DownloadButton imageRef="imageRef" />)
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
