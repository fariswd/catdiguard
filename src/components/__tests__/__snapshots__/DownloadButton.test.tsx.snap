// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download Button Component Should render without crashing 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#666",
          "fontSize": 28,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ïŠŠ
  </Text>
</View>
`;
